### PR TITLE
AUTH-337: PSa: add test for SCC-mutated PodSpec extraction

### PR DESCRIPTION
**Pull request by stlaz**
This adds tests for https://github.com/openshift/kubernetes/pull/1393

/assign @deads2k 
/cc @ibihim

### CODE REVIEW

File: test/extended/authorization/podsecurity_admission.go [hunk 1]
```diff
@@ -25,16 +40,185 @@ var _ = g.Describe("[sig-auth][Feature:PodSecurity]", func() {
...
+			name           string
+			container      corev1.Container
+			saName         string
+			expectWarnings bool
+		}{
+			{
+				name:           "SHOULD throw warnings if no SCC matches the pod template",
+				container:      privilegedContainer(sleeperContainer),
+				saName:         "default",
+				expectWarnings: false, // the PodSpecExtractor currently fails in this case and so nothing is thrown/logged in audits
```
Comment (stlaz) [inline test/extended/authorization/podsecurity_admission.go:115]: @deads2k this is due to https://github.com/openshift/kubernetes/commit/6fe5c8f28c24dc9de63cf54138a7a1adc3bc3b78#diff-7db8f52771920b1336fa638b8e08703795c29e33d6c242f7246f707ca3cb3048R99

Do we want to ignore workloads that don't match an SCC for audit/warnings?

**Comment by stlaz** [comment]
/retest-required
no related failures

File: test/extended/authorization/podsecurity_admission.go (hunk 1 above)
Comment (deads2k) [reply test/extended/authorization/podsecurity_admission.go:115]: So no SCC matched, we returned an error, the pod would not validate, and no warning appeared?

File: test/extended/authorization/podsecurity_admission.go [hunk 2]
```diff
@@ -25,16 +40,185 @@ var _ = g.Describe("[sig-auth][Feature:PodSecurity]", func() {
...
+			},
+			{
+				name:           "SHOULD throw warnings if SCC matches but will not mutate to the expected PSa profile",
+				container:      privilegedContainer(sleeperContainer),
+				saName:         privilegedSA.Name,
+				expectWarnings: true,
```
Comment (deads2k) [inline test/extended/authorization/podsecurity_admission.go:121]: be specific about the sub-strings expected.

File: test/extended/authorization/podsecurity_admission.go (hunk 1 above)
Comment (stlaz) [reply test/extended/authorization/podsecurity_admission.go:115]: No SCC matched, we returned an error.

The admission exits at that point (https://github.com/kubernetes/kubernetes/blob/03b78f3a15a69d72efd546ed867a46799f196a4b/staging/src/k8s.io/pod-security-admission/admission/admission.go#L435-L444), possibly waiting for the actual pod to appear.

No warnings are returned.

**Comment by stlaz** [comment]
/test all
CI appears to be stuck

**Review by ibihim — APPROVED** [review]
/lgtm

**Comment by stlaz** [comment]
/hold cancel
/retest required
nothing caused by this PR

**Comment by stlaz** [comment]
/retest-required
different, unrelated failures

**Comment by deads2k** [comment]
/hold cancel
/override ci/prow/e2e-gcp-ovn-upgrade 

akamai and service load balacner.

**Comment by stlaz** [comment]
/retest-required
/hold cancel
it looks to me as if though all the required tests passed already?

**Pull request merged**